cockpit/gui/mainWindow.py

```diff
@@ -351,35 +360,95 @@ def onViewLastFile(self, event = None):
 
 
+    ##user defined modes which include cameras and lasers active,
+    ##filter whieels etc...
+    def setMode(self, name):
+        #store mode to text file
+        if name == 'Save...':
+            self.onSaveExposureSettings(self.currentMode)
+        elif name == 'Load...':
+            self.onLoadExposureSettings()
+        elif name == 'Update' and self.currentMode != None:
+            events.publish('save exposure settings',
+                           self.storedModes[self.currentMode])
+            self.storeButton.setOption(self.currentMode)
+        elif name == 'New...':
+            self.createNewMode()
+        else:
+            print ('selected mode ',name) 
+            events.publish('load exposure settings', self.storedModes[name])
+            self.currentMode = name
+            self.storeButton.setOption(name)
+
+    def createNewMode(self):
+        #get name for new mode
+        # abuse get value dialog which will also return a string. 
+        value = cockpit.gui.dialogs.getNumberDialog.getNumberFromUser(
+            parent=self.topPanel, default='', title='New Mode Name',
+            prompt='Name')
+        self.storedModes[value]=dict()
+        #publish an event to populate mode settings.
+        events.publish('save exposure settings', self.storedModes[value])
+        #update button entries.
+        self.storeButton.setOptions(map(lambda name: (name,
+                                                       lambda n=name:
+                                                       self.setMode(n)),
+                                         self.storedModes))
+        #and set button value. 
+        self.storeButton.setOption(value)
+        self.currentMode = value
+
+                       
+                
     ## User wants to save the current exposure settings; get a file path
     # to save to, collect exposure information via an event, and save it.
-    def onSaveExposureSettings(self, event = None):
+    def onSaveExposureSettings(self, name, event = None):
         dialog = wx.FileDialog(self, style = wx.FD_SAVE, wildcard = '*.txt',
+                               defaultFile=name+'.txt',
                 message = "Please select where to save the settings.",
                 defaultDir = cockpit.util.user.getUserSaveDir())
         if dialog.ShowModal() != wx.ID_OK:
             # User cancelled.
+            self.storeButton.setOption(name)
             return
         settings = dict()
         events.publish('save exposure settings', settings)
         handle = open(dialog.GetPath(), 'w')
         handle.write(json.dumps(settings))
         handle.close()
+        self.storeButton.setOption(name)
 
     
     ## User wants to load an old set of exposure settings; get a file path
     # to load from, and publish an event with the data.
     def onLoadExposureSettings(self, event = None):
+        import os.path
```

Comment:
just a general and non-optional import should be done at top of the file

Reply:
Sure, ok, I'll move it